File: java/src/test/java/money_problem/domain/MoneyTest.java
package money_problem.domain;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static money_problem.domain.Currency.EUR;
import static money_problem.domain.Currency.KRW;
import static org.assertj.core.api.Assertions.assertThat;

class MoneyTest {
    @Test
    @DisplayName("10 EUR x 2 = 20 EUR")
    void shouldMultiplyInEuros() {
        assertThat(new Money(10, EUR).times(2))
                .isEqualTo(new Money(20, EUR));
    }

    @Test
    @DisplayName("4002 KRW / 4 = 1000.5 KRW")
    void shouldDivideInKoreanWons() {
        assertThat(new Money(4002, KRW).divide(4))
                .isEqualTo(new Money(1000.5, KRW));
    }
}

File: c#/money-problem.Tests/MoneyTest.cs
using FluentAssertions;
using money_problem.Domain;
using Xunit;

namespace money_problem.Tests;

public class MoneyTest
{
    [Fact(DisplayName = "10 EUR x 2 = 20 EUR")]
    public void MultiplyInEuros()
    {
        10d.Euros()
            .Times(2)
            .Should()
            .Be(20d.Euros());
    }

    [Fact(DisplayName = "4002 KRW / 4 = 1000.5 KRW")]
    public void DivideInKoreanWons()
    {
        4002d.KoreanWons()
            .Divide(4)
            .Should()
            .Be(1000.5d.KoreanWons());
    }
}